src/twisted/mail/smtp.py

```diff
@@ -1791,7 +1826,7 @@ class SenderMixin:
     def getMailFrom(self):
         if not self.done:
             self.done = 1
-            return str(self.factory.fromEmail)
+            return unicode(self.factory.fromEmail)
```

Comment:
This is going to return the repr of `b'some@address.com'` on Python 3, and decode `self.factory.fromEmail` as ASCII on Python 2.  This should probably be `unicode(self.factory.fromEmail, encoding=self._encoding)`.

Reply:
That doesn't work:

```
          File "/Users/crodrigues/twisted_15/src/twisted/mail/smtp.py", line 1829, in getMailFrom
            return unicode(self.factory.fromEmail, encoding=self._encoding)
        builtins.TypeError: decoding to str: need a bytes-like object, Address found
```